components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint/src/main/webapp/basicauth.jsp

```diff
@@ -38,20 +38,37 @@
 
 <jsp:directive.include file="init-loginform-action-url.jsp"/>
 
-
+<script src="libs/jquery_1.11.3/jquery-1.11.3.js"></script>
 <script>
-        function submitCredentials (e) {
-            e.preventDefault();
-            var userName = document.getElementById("username");
-            userName.value = userName.value.trim();
-            if(userName.value){
-                document.getElementById("loginForm").submit();
-            }
-        }
+    function goBack() {
+        window.history.back();
+    }
+    
+    // Handle form submission preventing double submission.
+    $(document).ready(function(){
+        $.fn.preventDoubleSubmission = function() {
+            $(this).on('submit',function(e){
+                var $form = $(this);
+                if ($form.data('submitted') === true) {
+                    // Previously submitted - don't submit again.
+                    e.preventDefault();
+                    console.warn("Prevented a possible double submit event");
+                } else {
+                    e.preventDefault();
+                    // Mark it so that the next submit can be ignored.
+                    $form.data('submitted', true);
```

Comment:
This would cause an issue if someone tries to submit with an empty URL. In the next time, the user will not be allowed to submit the form since it is already marked as submitted.

Reply:
+1. The issue actually happens when the submit button is clicked without a value in the username field. Fixed in https://github.com/wso2/carbon-identity-framework/pull/2371